cli.js

```diff
@@ -36,117 +33,64 @@ const cli = meow(`
 	string: [
 		'plugin',
 		'optimizationLevel',
+		'outdir',
 		'install'
 	],
 	alias: {
+		O: 'optimizationLevel',
 		P: 'plugin',
 		i: 'interlaced',
-		o: 'optimizationLevel',
+		o: 'outdir',
 		p: 'progressive'
 	}
 });
 
-function isFile(path) {
-	if (/^[^\s]+\.\w*$/.test(path)) {
-		return true;
-	}
-
-	try {
-		return fs.statSync(path).isFile();
-	} catch (err) {
-		return false;
-	}
-}
-
 const DEFAULT_PLUGINS = [
 	'gifsicle',
-	'jpegtran',
+	'mozjpeg',
 	'optipng',
 	'svgo'
 ];
 
-function run(src, dest) {
-	const plugins = cli.flags.plugin ? arrify(cli.flags.plugin) : DEFAULT_PLUGINS;
-	const imagemin = new Imagemin().src(src);
+const requirePlugins = (plugins, opts) => plugins.map(x => {
+	try {
+		x = require(`imagemin-${x}`)(opts);
```

Comment:
Sounds like the rule could be improved. There's no way for it to not return because of the `process.exit()`. It should detect that and ignore. Mind opening an ESLint ticket?

This should be possible without a violation:

```js
arr.map(() => {
	try {
		return require('foo');
	} catch (err) {
		console.error(err);
		process.exit(1);
	}
});
```